lib/go-tc/constants.go

```diff
@@ -19,28 +19,52 @@ package tc
  * under the License.
  */
 
+// The ErrorConstant type is used for error messages.
 type ErrorConstant string
 
+// Error converts ErrorConstants to a string.
 func (e ErrorConstant) Error() string { return string(e) }
 
+// DBError is an error message for database errors.
 const DBError = ErrorConstant("database access error")
+
+// NilTenantError is an error message used when the TentantID for a tenant that
+// owns the origin is nil.
```

Comment:
`NilTenantError` itself is only used in [origin.checkTenancy()](https://github.com/apache/trafficcontrol/blob/RELEASE-4.0.0-RC1/traffic_ops/traffic_ops_golang/origin/origins.go#L233).

Reply:
weird

Reply:
That said, TODeliveryServiceRequest [requires](https://github.com/apache/trafficcontrol/blob/RELEASE-4.0.0-RC1/traffic_ops/traffic_ops_golang/deliveryservice/request/requests.go#L165-L168) requires a DeliveryService's TenantID to be non-nil [when tenancy is enabled](https://github.com/apache/trafficcontrol/blob/RELEASE-4.0.0-RC1/traffic_ops/traffic_ops_golang/api/shared_handlers.go#L208-L210), so that's another place where `NilTenantError` could be used.

`DeliveryServiceTenantInfo`'s TenantID [also must be non-nil](https://github.com/apache/trafficcontrol/blob/RELEASE-4.0.0-RC1/traffic_ops/traffic_ops_golang/tenant/tenancy.go#L42-L44), but since the TenantID is not necessarily supplied from the request, `NilTenantError` doesn't apply here in its current wording.

Reworded `NilTenantError`'s comment in 5018d3a8df.